### DISCUSSION TITLE
Remove reCaptcha screen in mobile devices during Phone Authentication.

### DISCUSSION THREAD

**Discussion by pkitatta** [firebase/flutterfire#5114] [Q&A]
Phone auth has been working well in mobile devices, even in emulator, without launching recapture screen. Recently when I upgraded flutter and all the dependencies, phone auth started launching recapture browser in all devices.

The whole recaptcha process is a bit of an inconvenience and beats the simplicity of why we choose phone auth in the first place.
Recaptcha also prevents the normal authentication on phone with RAM that is below 1GB. This is because every time the browser is launched, the flutter app is pushed and killed in the background, and this cycle keeps repeating without letting the user signin.

I have followed the instructions in this video: https://www.youtube.com/watch?v=opBwTxicw1k, but I have failed to remove remove recapture on mobile devices.
Steps in video:
1. Add SHA-256
2. Enable Android Device Verification API in Project

I also get this in the console: SafetyNet Attestation fails basic integrity.

Question: Is there another way to remove recaptcha on devices?

Flutter doctor:
`Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 1.22.5, on Microsoft Windows [Version 10.0.18362.1256], locale en-US)

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[√] Android Studio (version 3.2)
[√] Connected device (1 available)

• No issues found!`

Relevant Dependencies:
`firebase_core: ^0.5.3
  firebase_auth: ^0.18.4+1
  firebase_messaging: ^7.0.3
  cloud_firestore: ^0.14.4`

**Comment by Vitor-Bukovitz** [comment]
This is actually not a problem with this plugin, but with the firebase phone auth itself.

**Reply by pkitatta** [reply]
Do you have an idea how to solve it?

**Reply by Vitor-Bukovitz** [reply]
@pkitatta  Can you try to open an issue on this repo?
https://github.com/firebase/firebase-android-sdk/issues

Also, I would like to upvote if you do.

**Reply by pkitatta** [reply]
@Vitor-Bukovitz I have opened issue there. Please upvote.

https://github.com/firebase/firebase-android-sdk/issues/2477

**Comment by the-Sreejith** [comment]
So I migrated my project from
```
  firebase_auth: ^0.18.1
  firebase_core: ^0.5.0
  cloud_firestore: ^0.14.3
  firebase_messaging: ^7.0.3
```

to 
```
  # firebase_core: "^0.7.0"
  # firebase_auth: "^0.20.0"
  # cloud_firestore: "^0.16.0"
  # firebase_messaging: "^8.0.0-dev.14"
```
then I am getting problems like reCaptcha, auto retrieval not working, etc, even if  I did all the things said in the above video that you have provided the link. To confirm the migration was the problem, I rolled back my changes and runner the app again and I can't see the reCaptcha or the auto retrieval problem. Am I missing something, Is there something that I am missing in the Gradle files or the authentication code.

**Reply by pkitatta** [reply]
I am also seeking help my friend.

**Reply by pkitatta** [reply]
@Sreejithns2002 I have opened issue there. Please upvote.

firebase/firebase-android-sdk#2477
